boot/zephyr/include/flash_map/flash_map.h

```diff
@@ -43,14 +43,64 @@ extern "C" {
  */
 #include <inttypes.h>
 
+/**
+ * @brief Structure describing an area of memory-mapped flash.
```

Comment:
This API should work fine for non memory-mapped flash. I don't think that is where the 'map' in the name comes from, it's more like just an alternate word chosen instead of 'partition'.

We do want to support non memory-mapped flash, and other than jumping to the code, it should currently work. Imagine an external SPI flash to hold slot 1 and scratch so that most of the flash could be used for the app.